osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs

```diff
@@ -546,7 +546,11 @@ protected void ApplyResult(Action<JudgementResult> application)
             // Ensure that the judgement is given a valid time offset, because this may not get set by the caller
             var endTime = HitObject.GetEndTime();
 
-            Result.TimeOffset = Math.Min(HitObject.HitWindows.WindowFor(HitResult.Miss), Time.Current - endTime);
+            Result.TimeOffset = Time.Current - endTime;
+
+            double missWindow = HitObject.HitWindows.WindowFor(HitResult.Miss);
+            if (missWindow > 0)
+                Result.TimeOffset = Math.Min(Result.TimeOffset, missWindow);
```

Comment:
A little uneasy about this change, it feels like it could break a few places - although I've checked a few off the top of my head and couldn't find anything so far, so maybe fine to go ahead with it.